### PR TITLE
Update get schema methods

### PR DESCRIPTION
**Pull request by SferaDev**
Re-use the logic we have right now in the frontend for arbitrary type mapping to a Xata type. We will inverse this soon, but for now make sure everything works as expected in pgroll branches

### CODE REVIEW

File: cli/src/commands/pull/pull.test.ts [hunk 1]
```diff
@@ -47,7 +47,77 @@ const baseFetch = (url: string, request: any) => {
...
+              columns: {
+                a: {
+                  name: 'a',
+                  type: 'string',
+                  default: null,
+                  nullable: true,
+                  unique: false,
+                  comment: ''
+                },
+                _createdat: {
```
Comment (eemmiillyy) [inline cli/src/commands/pull/pull.test.ts:70]: These internal fields will be renamed to `xata_` soon so we might want to already rename them to the new format?

File: cli/src/commands/push/push.test.ts [hunk 2]
```diff
@@ -62,7 +62,77 @@ const baseFetch = (url: string, request: any) => {
...
+              columns: {
+                a: {
+                  name: 'a',
+                  type: 'string',
+                  default: null,
+                  nullable: true,
+                  unique: false,
+                  comment: ''
+                },
+                _createdat: {
```
Comment (eemmiillyy) [inline cli/src/commands/push/push.test.ts:85]: Same comment as above

File: cli/src/migrations/pgroll.ts [hunk 3]
```diff
@@ -29,3 +30,86 @@ export async function allMigrationsPgRollFormat() {
...
+    case 'xata_file_array':
+      return 'file[]';
+    case 'real[]':
+      return 'vector';
+    default:
+      return type;
+  }
+}
+
+export async function getBranchDetailsWithPgRoll(
```
Comment (eemmiillyy) [inline cli/src/migrations/pgroll.ts:74]: By inverting the logic do you mean that this function will be exported in client? Or just the `pgRollToXataColumnType` mapping maybe?

**Review by eemmiillyy — APPROVED** [review]

File: cli/src/migrations/pgroll.ts (hunk 3 above)
Comment (SferaDev) [reply cli/src/migrations/pgroll.ts:74]: Ideally none, let's see in future PRs how we handle it

File: cli/src/commands/pull/pull.test.ts (hunk 1 above)
Comment (SferaDev) [inline cli/src/commands/pull/pull.test.ts:70]:
```suggestion
                xata_createdat: {
```

File: cli/src/commands/pull/pull.test.ts [hunk 4]
```diff
@@ -47,7 +47,77 @@ const baseFetch = (url: string, request: any) => {
...
+                  name: '_createdat',
+                  type: 'timestamptz',
+                  default: 'now()',
+                  nullable: false,
+                  unique: false,
+                  comment: ''
+                },
+                _id: {
```
Comment (SferaDev) [inline cli/src/commands/pull/pull.test.ts:78]:
```suggestion
                xata_id: {
```

File: cli/src/commands/pull/pull.test.ts [hunk 5]
```diff
@@ -47,7 +47,77 @@ const baseFetch = (url: string, request: any) => {
...
+                  name: '_id',
+                  type: 'text',
+                  default: null,
+                  nullable: false,
+                  unique: true,
+                  comment: ''
+                },
+                _updatedat: {
```
Comment (SferaDev) [inline cli/src/commands/pull/pull.test.ts:86]:
```suggestion
                xata_updatedat: {
```

File: cli/src/commands/pull/pull.test.ts [hunk 6]
```diff
@@ -47,7 +47,77 @@ const baseFetch = (url: string, request: any) => {
...
+                  name: '_updatedat',
+                  type: 'timestamptz',
+                  default: 'now()',
+                  nullable: false,
+                  unique: false,
+                  comment: ''
+                },
+                _version: {
```
Comment (SferaDev) [inline cli/src/commands/pull/pull.test.ts:94]:
```suggestion
                xata_version: {
```

File: cli/src/commands/push/push.test.ts (hunk 2 above)
Comment (SferaDev) [inline cli/src/commands/push/push.test.ts:85]:
```suggestion
                xata_createdat: {
```

File: cli/src/commands/push/push.test.ts [hunk 7]
```diff
@@ -62,7 +62,77 @@ const baseFetch = (url: string, request: any) => {
...
+                  name: '_createdat',
+                  type: 'timestamptz',
+                  default: 'now()',
+                  nullable: false,
+                  unique: false,
+                  comment: ''
+                },
+                _id: {
```
Comment (SferaDev) [inline cli/src/commands/push/push.test.ts:93]:
```suggestion
                xata_id: {
```

File: cli/src/commands/push/push.test.ts [hunk 8]
```diff
@@ -62,7 +62,77 @@ const baseFetch = (url: string, request: any) => {
...
+                  name: '_id',
+                  type: 'text',
+                  default: null,
+                  nullable: false,
+                  unique: true,
+                  comment: ''
+                },
+                _updatedat: {
```
Comment (SferaDev) [inline cli/src/commands/push/push.test.ts:101]:
```suggestion
                xata_updatedat: {
```

File: cli/src/commands/push/push.test.ts [hunk 9]
```diff
@@ -62,7 +62,77 @@ const baseFetch = (url: string, request: any) => {
...
+                  comment: ''
+                }
+              },
+              indexes: {
+                'bb_hmtsb6hnd552p1rencda7oo3eg_3hae5b._pgroll_new_foo_pkey': {
+                  name: 'bb_hmtsb6hnd552p1rencda7oo3eg_3hae5b._pgroll_new_foo_pkey',
+                  unique: true,
+                  columns: ['_id']
+                }
+              },
```
Comment (SferaDev) [inline cli/src/commands/push/push.test.ts:124]:
```suggestion
              indexes: {},
```

File: cli/src/commands/push/push.test.ts [hunk 10]
```diff
@@ -62,7 +62,77 @@ const baseFetch = (url: string, request: any) => {
...
+              primaryKey: ['_id'],
```
Comment (SferaDev) [inline cli/src/commands/push/push.test.ts:125]:
```suggestion
              primaryKey: ['xata_id'],
```

File: cli/src/commands/pull/pull.test.ts [hunk 11]
```diff
@@ -47,7 +47,77 @@ const baseFetch = (url: string, request: any) => {
...
+                  comment: ''
+                }
+              },
+              indexes: {
+                'bb_hmtsb6hnd552p1rencda7oo3eg_3hae5b._pgroll_new_foo_pkey': {
+                  name: 'bb_hmtsb6hnd552p1rencda7oo3eg_3hae5b._pgroll_new_foo_pkey',
+                  unique: true,
+                  columns: ['_id']
+                }
+              },
```
Comment (SferaDev) [inline cli/src/commands/pull/pull.test.ts:109]:
```suggestion
              indexes: {},
```

File: cli/src/commands/pull/pull.test.ts [hunk 12]
```diff
@@ -47,7 +47,77 @@ const baseFetch = (url: string, request: any) => {
...
+              primaryKey: ['_id'],
```
Comment (SferaDev) [inline cli/src/commands/pull/pull.test.ts:110]:
```suggestion
              primaryKey: ['xata_id'],
```

**Pull request merged**